File: Patterns/10. Hourglass Patterns/96. Hourglass Even Numbers/hen.py
n = int(input("Enter a Number : "))

for i in range(1,2*n):

    row = min(i,2*n-i)
    
    # print spaces
    for j in range(1,row):
        print(end='   ')
    # print first half
    for j in range(1,n-row+2):
        print(f"{2*j:3}", end='')
    # print second half
    for j in range(n-row,0,-1):
        print(f"{2*j:3}", end='')
    print()


'''
Output:

  2  4  6  8 10  8  6  4  2
     2  4  6  8  6  4  2
        2  4  6  4  2
           2  4  2
              2
           2  4  2
        2  4  6  4  2
     2  4  6  8  6  4  2
  2  4  6  8 10  8  6  4  2  


'''

File: Patterns/10. Hourglass Patterns/96. Hourglass Even Numbers/hen.cpp
#include <iostream>
#include <iomanip>
#include <algorithm>
using namespace std;
int main(){
    int n,row;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=2*n-1; i++){

        row = min(i,2*n-i);

        // print spaces
        for(int j=1; j<=row-1; j++){
            cout<<"   ";
        }
        // print first half
        for(int j=1; j<=n-row+1; j++){
            cout<<setw(3)<<2*j;
        }
        // print second half
        for(int j=n-row; j>=1; j--){
            cout<<setw(3)<<2*j;
        }
        cout<<'\n';

    }
}

/*
Output

  2  4  6  8 10  8  6  4  2
     2  4  6  8  6  4  2
        2  4  6  4  2
           2  4  2
              2
           2  4  2
        2  4  6  4  2
     2  4  6  8  6  4  2
  2  4  6  8 10  8  6  4  2

*/